test/blackbox-tests/test-cases/coq/failed-config.t/run.t

```diff
@@ -47,11 +48,36 @@ Now we create a simple project that uses this coqc wrapper.
   >   (write-file a.v "")))
   > EOF
 
+Should fail: first warning that installed theories are being skipped due to the
+failure, then, as the library requires the stdlib, it fails:
+  $ FAIL_CONFIG=1 \
+  > dune build
+  Warning: Skipping installed theories due to 'coqc --config' failure:
+  - $TESTCASE_ROOT/bin/coqc --config failed with exit code 1.
+  Hint: Try running 'coqc --config' manually to see the error.
+  Error: Couldn't find Coq standard library, and theory is not using (stdlib
+  no)
+  [1]
+
 Here we build a simple Coq project. Neither a failing --config or --print-version should
-block this.
+block this as it doesn't depend on the stdlib.
+
+  $ cat > dune <<EOF
+  > (coq.theory
+  >  (flags -noinit)
+  >  (name foo)
+  >  (stdlib no))
+  > 
+  > (rule
+  >  (deps
+  >   (env_var FAIL_VERSION)
+  >   (env_var FAIL_CONFIG))
+  >  (action
+  >   (write-file a.v "")))
+  > EOF
 
-Should succeed, but should warn that installed theories are being skipped due to the
-failure.
+Should succeed, warning that installed theories are being skipped due to the
+failure (c.f. #8958):
```

Comment:
@Alizter , so we keep this test, as we should, but only for `(stdlib no)`, we add a new test (just above) that correctly fails when no stdlib has been detected.

Note that the behavior you propose (if no stdlib is found, default to implicit) is already the default but only for '(lang coq)` < 0.8